pkg/cluster/internal/providers/docker/provision.go

```diff
@@ -255,6 +255,20 @@ func runArgsForNode(node *config.Node, clusterIPFamily config.ClusterIPFamily, n
 		args = append(args, "-e", "KUBECONFIG=/etc/kubernetes/admin.conf")
 	}
 
+	// Append CDI device args (used for GPU support)
+	if len(node.CDIDevices) > 0 {
+		// Check for docker > 25
+		ver := Version()
+		if ver != "dev" || strings.Split(ver, ".")[0] < "25" {
+			return nil, errors.Errorf("using devices api in kind requires Docker >= v25, but found %q", ver)
+		}
+
+		// Append args for each device
+		for _, device := range node.CDIDevices {
+			args = append(args, "--device", strings.TrimSpace(device))
+		}
+	}
+
```

Comment:
Will also allow passing "standard" devices via the `CDIDevices` setting? It looks like there is a new `validate()` call that will prevent this though. Is the validate call made sometime before this is done here?

Reply:
I am using `parser.ParseQualifiedName(device)` instead of `parser.ParseQualifiedName(device)`. The results are the same except you get the additional error message about why your device string is invalid. I think this is preferable. ParseQualifiedName() is called in IsQualifiedName().

This doesn't change the dependencies at all. There are tests on `github.com/container-orchestrated-devices/container-device-interface/pkg/parser` pulling in the additional packages. Maybe we should put the tests in a different package?